### PR TITLE
Branch Parser Update

### PR DESCRIPTION
**Pull request by t-l-xin**
Edit ParserTest to check for return Command type
Add assert & logging in functional code

### CODE REVIEW

File: src/test/java/seedu/duke/ParserTest.java [hunk 1]
```diff
@@ -52,162 +80,172 @@ public void parse_NotProgramCommand_errorMessage() {
...
+    public void parse_addCommandValidArgs_returnsAddCommand() throws ItemNotExistException,
+            NoPropertyFoundException, IllegalFormatException {
+
+        String input1 = ADD_STRING + " n/" + ITEM_NAME_EXAMPLE_1 + " c/" + CATEGORY_EXAMPLE_1 + " p/$"
+                + PURCHASE_COST_EXAMPLE_1 + " s/$" + SELLING_PRICE_EXAMPLE_1 + " q/" + QUANTITY_EXAMPLE_1;
+
+        AddCommand expectedCommand1 = new AddCommand(ITEM_NAME_EXAMPLE_1,
+                PURCHASE_COST_EXAMPLE_1, SELLING_PRICE_EXAMPLE_1);
+
+        assertEquals(expectedCommand1.getClass(), parser.parseCommand(input1, list).getClass());
```
Comment (haoyusimon) [inline src/test/java/seedu/duke/ParserTest.java:110]: Not sure if it is enough to just check that the Command returned by the parseCommand is of the same Command type with the expected Command. I was thinking if can add an equals() method in command to compare the commands in a more accurate way. But maybe can improve on the testing in a later version.

File: src/main/java/seedu/duke/Duke.java [hunk 2]
```diff
@@ -1,21 +1,23 @@
 package seedu.duke;
 
+import seedu.duke.command.Command;
+import seedu.duke.model.ItemContainer;
+import seedu.duke.model.exception.IllegalArgumentException;
```
Comment (yuejunfeng0909) [inline src/main/java/seedu/duke/Duke.java:5]: i think dont need to import these

File: src/main/java/seedu/duke/parser/Parser.java [hunk 3]
```diff
@@ -72,33 +74,38 @@
...
      * respective cases depending on the command word.
      *
-     * @param userInputLine The user input Line
-     * @param list The itemContainer used to prepare the command
-     * @return A string indicating parse success or failure
+     * @param userInputLine The user input Line.
+     * @param list          The itemContainer used to prepare the command.
+     * @return Command object depending on the command type.
+     * @throws IllegalFormatException   If user input line does not match the respective command format.
+     * @throws ItemNotExistException    If item name not found in the container.
```
Comment (yuejunfeng0909) [inline src/main/java/seedu/duke/parser/Parser.java:92]: I think its better not to leave full stop here, according to https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/documentation.html#javadoc

File: src/test/java/seedu/duke/ParserTest.java (hunk 1 above)
Comment (t-l-xin) [reply src/test/java/seedu/duke/ParserTest.java:110]: Noted, will discuss more in the later parts.

File: src/main/java/seedu/duke/parser/Parser.java (hunk 3 above)
Comment (t-l-xin) [reply src/main/java/seedu/duke/parser/Parser.java:92]: Noted

**Pull request merged**